sonic_installer/bootloader/bootloader.py

```diff
@@ -43,8 +52,23 @@ def verify_binary_image(self, image_path):
         """verify that the image is supported by the bootloader"""
         raise NotImplementedError
 
+    def verify_reboot(self):
+        """verify the image for reboot"""
+        image = self.get_next_image()
+        image_path = self.get_image_path(image)
+        if not path.exists(image_path):
+            sys.stderr.write('Next image {0} doesn\'t exist ...\n'.format(image))
```

Comment:
This function shouldn't write to stderr. I think the calling application should be responsible for outputting to stderr or osyslog based on the return value.

Reply:
Fixed.

Reply:
I agree. Should not print to either stderr or stdout.
This function is responsible to verify and return Boolean.
The caller is responsible to interpret the Boolean return value.

---
In reply to: [452556977](https://github.com/Azure/sonic-utilities/pull/979#discussion_r452556977) [](ancestors = 452556977)

Reply:
The message removed.